ananas/src/main/java/iamutkarshtiwari/github/io/ananas/editimage/adapter/FilterAdapter.java

```diff
@@ -54,10 +59,31 @@ public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int pos) {
         filterViewHolder.text.setText(name);
 
         String imageUrl = "drawable/" + filterImages[position];
-        int imageKey = filterListFragment.getResources().getIdentifier(imageUrl, "drawable", context.getPackageName());
-        filterViewHolder.icon.setImageDrawable(filterListFragment.getResources().getDrawable(imageKey));
+        int imageKey = filterListFragment.getResources()
+                .getIdentifier(imageUrl, "drawable", context.getPackageName());
+        filterViewHolder.icon.setImageDrawable(ContextCompat.getDrawable(context, imageKey));
 
         filterViewHolder.icon.setTag(position);
-        filterViewHolder.icon.setOnClickListener(v -> filterListFragment.enableFilter(position));
+        filterViewHolder.icon.setOnClickListener(v -> {
+            if (holder.getAdapterPosition() == RecyclerView.NO_POSITION) return;
+
+            notifyItemChanged(selectedPosition);
+            selectedPosition = position;
+            notifyItemChanged(selectedPosition);
+
+            filterListFragment.enableFilter(position);
+        });
+
+        if (selectedPosition == pos) {
```

Comment:
should this be holder.getAdapterPosition instead?
I remember there's a caveat around using the given position

Reply:
I'm not sure about this one. I thought, I'm using the position given by the `onBindViewHolder`. In case if I should user adapter position then I should change everything within `onBindViewHolder`